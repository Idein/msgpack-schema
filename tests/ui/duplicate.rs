use msgpack_schema::*;

#[derive(Serialize)]
struct S1 {
    #[optional]
    #[optional]
    #[tag = 1]
    x: Option<String>,
}

#[derive(Deserialize)]
struct S2 {
    #[optional]
    #[optional]
    #[tag = 1]
    x: Option<String>,
}

#[derive(Serialize)]
struct S3 {
    #[tag = 1]
    #[tag = 2]
    x: String,
}

#[derive(Deserialize)]
struct S4 {
    #[tag = 1]
    #[tag = 2]
    x: String,
}

#[derive(Serialize)]
#[untagged]
#[untagged]
struct S5 {}

#[derive(Deserialize)]
#[untagged]
#[untagged]
struct S6 {}

fn main() {}
