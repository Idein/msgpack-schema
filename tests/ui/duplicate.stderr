error: duplicate #[optional] attribute
 --> $DIR/duplicate.rs:6:5
  |
6 |     #[optional]
  |     ^^^^^^^^^^^

error: duplicate #[optional] attribute
  --> $DIR/duplicate.rs:14:5
   |
14 |     #[optional]
   |     ^^^^^^^^^^^

error: duplicate #[tag] attribute
  --> $DIR/duplicate.rs:22:5
   |
22 |     #[tag = 2]
   |     ^^^^^^^^^^

error: duplicate #[tag] attribute
  --> $DIR/duplicate.rs:29:5
   |
29 |     #[tag = 2]
   |     ^^^^^^^^^^

error: duplicate #[untagged] attribute
  --> $DIR/duplicate.rs:35:1
   |
35 | #[untagged]
   | ^^^^^^^^^^^

error: duplicate #[untagged] attribute
  --> $DIR/duplicate.rs:40:1
   |
40 | #[untagged]
   | ^^^^^^^^^^^
